arnparse/arnparse.py

```diff
@@ -45,3 +42,19 @@ def arnparse(arn_str):
         resource_type=resource_type,
         resource=resource,
     )
+
+
+def _parse_resource(resource):
+    first_separator_index = -1
+    for idx, c in enumerate(resource):
+        if c in (':', '/'):
+            first_separator_index = idx
+            break
+
+    if first_separator_index != -1:
+        c = resource[first_separator_index]
```

Comment:
Renaming `c` to `separator` would help understand the logic here.

Reply:
`c` is actually for `char`, so I'm not sure `separator` is a good name